addons/project/controllers/portal.py

```diff
@@ -224,6 +224,18 @@ def portal_my_project_task(self, project_id=None, task_id=None, access_token=Non
         task_sudo.attachment_ids.generate_access_token()
         values = self._task_get_page_view_values(task_sudo, access_token, project=project_sudo, **kw)
         values['project'] = project_sudo
+
+        history = request.session.get('my_project_tasks_history', [])
+        try:
+            current_task_index = history.index(task_id)
+        except ValueError:
+            return request.render("project.portal_my_task", values)
+
+        total_task = len(history)
+        task_url = f"/my/project/{project_id}/task/%s?model=project.project&res_id={values['user'].id}&access_token={access_token}"
+        values['prev_record'] = total_task > 1 and task_url % history[(current_task_index - 1) % total_task]
+        values['next_record'] = total_task > 1 and task_url % history[(current_task_index + 1) % total_task]
```

Comment:
IMO, this code could be moved in `_task_get_page_view_values` method and we will execute the code only if we have a project in that method. 🙂 

For the 2 last lines, first of all, good idea to improve the portal pager. 👍🏽 However, I don't agree to do that in stable version and only in tasks, it is not really a good idea to have a different behavior than the generic one. 😬  

I suggest to just fix the issue and keep the generic behavior in 15. In master, we could discuss with @tde-banana-odoo to know if we could do that in generic way too (more precisely, we will also do the changes in `portal` module). That is:
- when the record is the first one in the history then the previous button will return the last record in the history instead of disabling the button.
- when the record is the last one in the history then the next button will return the first record in the history instead of disabling the button.

cc @IT-Ideas and @MissingNoShiny